File: 2243-calculate-digit-sum-of-a-string/2243-calculate-digit-sum-of-a-string.py
class Solution:
    def digitSum(self, s: str, k: int) -> str:
        while len(s) > k:
            res = ''
            for i in range(0, len(s), k):
                slice = s[i:i + k]
                res += str(sum(int(char) for char in slice))
            s = res
            
        return s

File: 2243-calculate-digit-sum-of-a-string/2243-calculate-digit-sum-of-a-string.rb
# @param {String} s
# @param {Integer} k
# @return {String}
def digit_sum(s, k)
    until s.size <= k
        res=''
        s.chars.each_slice(k) do |slice|
            res += (slice.sum(&:to_i)).to_s
        end
        s=res
    end
    s
end
